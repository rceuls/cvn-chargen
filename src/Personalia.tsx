import { ICharacter } from './services/character.generator';
import { getSRDLink } from './services/util';

const Personalia = ({ character }: { character: ICharacter }) => {
  return (
    <>
      <div className="row">
        <div className="col-sm">
          You are {character.name} the{' '}
          <a
            href={getSRDLink(character.charClass.url)}
            target="_blank"
            rel="noreferrer"
          >
            {character.charClass.name}
          </a>
        </div>
      </div>
      {character.charClass.suffix ? (
        <div className="row">
          <div className="col-sm">{character.charClass.suffix}</div>
        </div>
      ) : (
        <></>
      )}
      <hr />
    </>
  );
};

export default Personalia;
