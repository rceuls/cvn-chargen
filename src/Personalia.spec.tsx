import React from 'react';
import renderer from 'react-test-renderer';
import Personalia from './Personalia';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Personalia
        character={{
          name: 'Jos',
          charClass: { suffix: 'ayy', name: 'test class', url: 'Moo' },
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
