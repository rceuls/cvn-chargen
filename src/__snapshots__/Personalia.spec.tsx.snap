// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="row"
  >
    <div
      className="col-sm"
    >
      You are 
      Jos
       the
       
      <a
        href="https://oldschoolessentials.necroticgnome.com/srd/index.php/Moo"
        rel="noreferrer"
        target="_blank"
      >
        test class
      </a>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-sm"
    >
      ayy
    </div>
  </div>,
  <hr />,
]
`;
